### ISSUE TITLE
Manual Code is 15 chars. Authenticator Plus says it's too short

### ISSUE THREAD

**Issue by gtmadev** [BrandonPotter/GoogleAuthenticator#14]
The code works well. I have also tested with a different QR code generator. But one thing I can't seem to figure out is why the manual code is not accepted in my 2FA app (Authenticator Plus). If I scan the code, it works fine. If I try to enter the code manually, it says it is too short.

I checked another app and I see that the manual code on that one is 16 characters. And that does work. So why is the manual code too short???

**Comment by mandy0928** [comment]
I'm having the same problem too can anyone tell me how to solve it?

**Comment by gtmadev** [comment]
I had to open up my solution to recall as I was just testing 2FA at the time. In my code, I see that I remarked out the code in question and installed another library (which did work). I never figured out why this one failed to work.

This one did work for me....
[https://github.com/RobThree/TwoFactorAuth.Net](url)

**Comment by ahwm** [comment]
I'm a little late here, but the length of the manual code is determined by the length of the secret key

A key such as `f68f1fe894` is only 10 characters and produces a setup code of 16 characters. I was able to successfully add an account to Authenticator Plus using this key and it's subsequent setup code of `MY3DQZRRMZSTQOJU`.

I was also able to set up an account using `f68f1` as the key (which resulted in a setup code of `MY3DQZRR`, which is only 8 characters). I did notice that if the setup code had equals on the end (Base32 padding) then Authenticator Plus didn't like them. I could add those to other apps (Google Authenticator, Authy, Microsoft Authenticator) just fine.
